src/services/expenses.service.js

```diff
@@ -0,0 +1,93 @@
+const { filterExpenses } = require('../helpers/filterExpenses');
+const { generateId } = require('../helpers/generateId');
+const { ERRORS } = require('../variables/variables');
+const { usersService } = require('./users.service');
+
+let expensesArray = [];
+
+const initExpensesService = () => {
+  expensesArray = [];
+};
+
+const getExpenses = (params) => {
+  return filterExpenses(expensesArray, params);
+};
+
+const createExpenses = (expenses) => {
+  const newExpenses = { ...expenses, id: generateId() };
+
+  const REQUIRED_FIELDS = ['userId', 'spentAt', 'title', 'amount', 'category'];
+
+  if (REQUIRED_FIELDS.some((field) => !expenses[field])) {
+    throw new Error(ERRORS.bodyRequired);
+  }
+
+  try {
+    usersService.getUserById(newExpenses.userId);
+
+    expensesArray.push(newExpenses);
+  } catch (err) {
+    throw new Error(ERRORS.badUserRequest);
+  }
```

Comment:
```suggestion
    usersService.getUserById(newExpenses.userId);

    expensesArray.push(newExpenses);
```
Why it's a bad request if some error occurs in the database?